src/sentry/utils/performance_issues/performance_detection.py

```diff
@@ -1383,14 +1383,14 @@ def on_complete(self):
                 )
 
     def _fingerprint(self, span_list) -> str:
-        call_stack_strings = []
-        overall_stack = []
+        call_stack_strings = set()
+        overall_stack = set()
```

Comment:
Will this ever cause an issue as sets don't preserve order?
I.e you'll create new fingerprints for issues that already exists.

Reply:
Oh good catch, i'll probably have to remove duplicates from the list instead of just using a set :+1:

Reply:
Yup! Or just check `if not in` before insertion

Reply:
ok but hear me out, what about:
`sorted(set(list), lambda x: list.index(x))`